### ISSUE TITLE
MemoizedInstanceArguments new cop

### ISSUE THREAD

**Issue by ngouy** [rubocop-hq/rubocop#8135]
## Is your feature request related to a problem? Please describe.

the other day by inadvertence I used memoization with an instance method which took some attributes and by so, may have been returned a different results depending on the given attributes.

simplified exemple
```rb
def sanitized_titles
  titles.map { |title| sanitized_title(title) }
end

def sanitized_title(title)
  @sanitized_title ||= title.strip
end
```

## Describe the solution you'd like

I'm almost sure it can be a bad idea for many corner case reasons, but I don't know them yet, that's why I give it a try.

I saw we have useful policies for memorized attributes. (which ensure var names are consistents for example)

What do you think about a policy which warn you when you use memoization for method which takes arguments?

## Describe alternatives you've considered

Not implement this policy cuz it's a bad idea, and be more careful when I use memoization ¯\\_(ツ)_/¯
